nhaystack-rt/src/nhaystack/server/TagManager.java

```diff
@@ -912,55 +912,63 @@ private void addSiteEquipTags(
         HDict tags)
     {
         // the point is explicitly tagged with an equipRef
-        // the hdb will contain an equipRef tag converted from a equipRef relation if it exist.
+        // the hdb will contain an equipRef tag converted from a equipRef
+        // relation, if it exists.
         if (hdb.has(EQUIP_REF))
         {
-            BComponent equip = lookupComponent((HRef) hdb.get(EQUIP_REF));
-
-            if (equip != null)
+            if (!hdb.has(SITE_REF))
             {
-                // this will convert an existing siteRef relation to a siteRef tag in thie HDict equipTags
-                final HDict equipTags = convertRelationsToRefTags(equip);
```

Comment:
This conversion has probably already happened so calling this again here causes another getAll relations.  If no siteRef relation is present in this method, the new findSiteRef method will do a more efficient and targeted search for siteRef relations instead doing a getAll relations.